testsuite/integration-arquillian/db-allocator-plugin/pom.xml

```diff
@@ -0,0 +1,86 @@
+~ and other contributors as indicated by the @author tags.
+~
+~ Licensed under the Apache License, Version 2.0 (the "License");
+~ you may not use this file except in compliance with the License.
+~ You may obtain a copy of the License at
+~
+~ http://www.apache.org/licenses/LICENSE-2.0
+~
+~ Unless required by applicable law or agreed to in writing, software
+~ distributed under the License is distributed on an "AS IS" BASIS,
+~ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+~ See the License for the specific language governing permissions and
+~ limitations under the License.
+-->
+
+<project xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd" xmlns="http://maven.apache.org/POM/4.0.0"
+         xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
+    <parent>
+        <groupId>org.keycloak.testsuite</groupId>
+        <artifactId>integration-arquillian</artifactId>
+        <version>6.0.0-SNAPSHOT</version>
+    </parent>
+    <modelVersion>4.0.0</modelVersion>
+
+    <artifactId>db-allocator-plugin</artifactId>
+    <packaging>maven-plugin</packaging>
+    <name>DB Allocator Plugin</name>
+
+    <properties>
+        <maven.version>3.6.0</maven.version>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-plugin-api</artifactId>
+            <version>${maven.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven.plugin-tools</groupId>
+            <artifactId>maven-plugin-annotations</artifactId>
+            <version>${maven.version}</version>
+            <scope>provided</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.apache.maven</groupId>
+            <artifactId>maven-core</artifactId>
+            <version>${maven.version}</version>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.resteasy</groupId>
+            <artifactId>resteasy-client</artifactId>
+        </dependency>
+
+        <dependency>
+            <groupId>junit</groupId>
+            <artifactId>junit</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.jboss.arquillian.container</groupId>
+            <artifactId>undertow-embedded</artifactId>
+            <scope>test</scope>
+        </dependency>
+        <dependency>
+            <groupId>org.keycloak.testsuite</groupId>
+            <artifactId>integration-arquillian-servers-app-server-undertow</artifactId>
+            <version>6.0.0-SNAPSHOT</version>
+            <scope>test</scope>
+        </dependency>
+
+    </dependencies>
+
+        <build>
+        <plugins>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-plugin-plugin</artifactId>
+                <version>3.6.0</version>
```

Comment:
Couldn't we use ${maven.version} here?

Reply:
I don't think this is a good idea to be honest. If I do that, the plugin compilation result might depend on your environment. 

Note, that Maven (at least the bits I used in the plugin) are backwards compatible. The plugin is compiled against `3.6.0` and I'm using `3.5.4` on my local box. Everything works fine with this setup.

Reply:
@vramik Could you please revisit this comment and tell me what you think?

Reply:
@slaskawi I've noticed there you've the property specified https://github.com/keycloak/keycloak/pull/5951/files#diff-b6a614e004ec399dc0a218c01b72965cR34 so my thinking was to use it. If there is a reason (and you described that it is) to remain current state, ignore my comment.

Reply:
Ok, thanks!

Reply:
Unfortunately it seems my approach was wrong. Since I'm using a more recent of Maven Core dependency, Maven freaked out when invoked on the prod pipeline. It seems using `${maven.version}` as you adviced might be a better choice (although we might run into some problems in the future, but I hope not :imp: ).